proto/rpc/v1/amount.proto

```diff
@@ -0,0 +1,35 @@
+syntax = "proto3";
+
+package rpc.v1;
+
+message CurrencyAmount {
+    oneof amount {
+        XRPAmount xrp_amount = 1;
+        FiatAmount fiat_amount = 2;
+    }
+}
+
+// A representation of an amount of XRP.
+message XRPAmount {
+    // A numeric string representing the number of drops of XRP.
+    uint64 drops = 1;
+}
+
+// A representation of an amount of currency in fiat.
+message FiatAmount {
+    // The currency used to value the amount.
+    Currency currency = 1;
+
+    // The value of the amount.
+    string value = 2;
+
+    // Unique account address of the entity issuing the currency.
+    string issuer = 3;
+}
+
+// Available currencies.
+message Currency {
+
+    string name = 1;
```

Comment:
Well, though _recommended_, the name doesn't have to be an ISO 4217 code; they can be any 3 ASCII characters, with some exceptions. So I hesitate to call it "ISO 4217 name" since they don't have to be an ISO 4217 Code. I recognize though that there might be a better name for the field than just "name".

Reply:
@MarkusTeufelberger I added comments to clarify, lmk what you think